### PR TITLE
signing key implementation

### PR DESCRIPTION
**Pull request by butonic**
use ocis-store to get or create a signing key

run https://github.com/owncloud/ocis-store/pull/2 and https://github.com/owncloud/ocis-ocs/pull/18
test using `curl localhost:9110/v2.php/cloud/user/signing-key -v`

related https://github.com/owncloud/ocis-proxy/pull/75

### CODE REVIEW

**Comment by butonic** [comment]
wtf https://cloud.drone.io/owncloud/ocis-store/11/3/3
```
1 | 1.13: Pulling from webhippie/golang
-- | --
2 | Digest: sha256:7e764a90464ce1a1fef182561189bdfa5260e653ebdd0d72b4b766550cb33d37
3 | Status: Image is up to date for webhippie/golang:1.13
+ make build
--
5 | protoc \
6 | -I=pkg/proto/v0/ \
7 | --go_out=pkg/proto/v0 store.proto
8 | protoc-gen-go: program not found or is not executable
9 | Please specify a program using absolute path or make sure the program is available in your PATH system variable
10 | --go_out: protoc-gen-go: Plugin failed with status code 1.
11 | make: *** [Makefile:165: pkg/proto/v0/store.pb.go] Error 1

```

**Comment by kulmann** [comment]
> wtf https://cloud.drone.io/owncloud/ocis-store/11/3/3
> 
> ```
> 
> 1 | 1.13: Pulling from webhippie/golang
> 
> -- | --
> 
> 2 | Digest: sha256:7e764a90464ce1a1fef182561189bdfa5260e653ebdd0d72b4b766550cb33d37
> 
> 3 | Status: Image is up to date for webhippie/golang:1.13
> 
> + make build
> 
> --
> 
> 5 | protoc \
> 
> 6 | -I=pkg/proto/v0/ \
> 
> 7 | --go_out=pkg/proto/v0 store.proto
> 
> 8 | protoc-gen-go: program not found or is not executable
> 
> 9 | Please specify a program using absolute path or make sure the program is available in your PATH system variable
> 
> 10 | --go_out: protoc-gen-go: Plugin failed with status code 1.
> 
> 11 | make: *** [Makefile:165: pkg/proto/v0/store.pb.go] Error 1
> 
> 
> 
> ```

Same as https://github.com/owncloud/ocis-settings/issues/30
Still not solved...

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/service.go  1
         

Clones removed
==============
+ pkg/service/v0/service.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-ocs/pullRequest?prid=5863812&bid=19195023)

File: pkg/service/v0/service.go [hunk 1]
```diff
@@ -7,10 +7,14 @@ import (
 	"github.com/go-chi/chi"
 	"github.com/go-chi/chi/middleware"
 	"github.com/go-chi/render"
+
+	//"github.com/micro/go-micro/v2"
```
Comment (refs) [inline pkg/service/v0/service.go:11]: spurious line

File: pkg/service/v0/service.go [hunk 2]
```diff
@@ -99,22 +103,34 @@ func (o Ocs) GetUser(w http.ResponseWriter, r *http.Request) {
 func (o Ocs) GetSigningKey(w http.ResponseWriter, r *http.Request) {
 
 	// TODO move token marshaling to ocis-proxy
```
Comment (refs) [inline pkg/service/v0/service.go:105]: is this TODO still relevant? if so, worth transforming it into an issue?

File: pkg/service/v0/service.go [hunk 3]
```diff
@@ -99,22 +103,34 @@ func (o Ocs) GetUser(w http.ResponseWriter, r *http.Request) {
...
+	_, ok := user.ContextGetUser(r.Context())
 	if !ok {
-		render.Render(w, r, ErrRender(MetaBadRequest.StatusCode, "missing user in context"))
-		return
+		//	render.Render(w, r, ErrRender(MetaBadRequest.StatusCode, "missing user in context"))
+		//	return
+	}
+	c := storepb.NewStoreService("com.owncloud.api.store", grpc.NewClient())
+	res, err := c.Read(r.Context(), &storepb.ReadRequest{
+		Key: "TODO replace with user from ctx",
```
Comment (refs) [inline pkg/service/v0/service.go:113]: wouldn't it be better to use the user's UUID from the account service? It should be present in the access token if I recall correctly

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/middleware/accesstoken.go  4
- pkg/service/v0/service.go  4
- pkg/middleware/options.go  2
         

Clones removed
==============
+ pkg/service/v0/service.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-ocs/pullRequest?prid=5863812&bid=19195023)

File: pkg/service/v0/service.go [hunk 4]
```diff
@@ -102,35 +108,60 @@ func (o Ocs) GetUser(w http.ResponseWriter, r *http.Request) {
...
 	c := storepb.NewStoreService("com.owncloud.api.store", grpc.NewClient())
 	res, err := c.Read(r.Context(), &storepb.ReadRequest{
-		Key: "TODO replace with user from ctx",
 		Options: &storepb.ReadOptions{
-			Database: "ocs",
+			Database: "proxy",
 			Table:    "signing-keys",
-		}})
+		},
+		Key: u.GetUsername(), // TODO username or id?
```
Comment (refs) [inline pkg/service/v0/service.go:124]: why not using the ocis account instead of the reva user?

Comment (butonic) [reply pkg/service/v0/service.go:124]: I think phoenix will send the username as the credential in signed requests on oc10 ... need to to get a phoenix with url signing enabled to test what happens on top of ocis.

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/middleware/accesstoken.go  4
- pkg/middleware/options.go  2
- pkg/service/v0/service.go  4
         

Clones removed
==============
+ pkg/service/v0/service.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-ocs/pullRequest?prid=5863812&bid=19195023)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/service.go  4
- pkg/middleware/options.go  2
- pkg/middleware/accesstoken.go  4
         

Clones removed
==============
+ pkg/service/v0/service.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-ocs/pullRequest?prid=5863812&bid=19195023)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/service.go  4
- pkg/middleware/options.go  2
- pkg/middleware/accesstoken.go  4
         

Clones removed
==============
+ pkg/service/v0/service.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-ocs/pullRequest?prid=5863812&bid=19195023)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/middleware/accesstoken.go  4
- pkg/service/v0/service.go  4
- pkg/middleware/options.go  2
         

Clones removed
==============
+ pkg/service/v0/service.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-ocs/pullRequest?prid=5863812&bid=19195023)

File: pkg/service/v0/service.go (hunk 2 above)
Comment (butonic) [reply pkg/service/v0/service.go]: yep, tracking in https://github.com/owncloud/ocis-ocs/issues/19 and removing the todo in the code

File: pkg/service/v0/service.go (hunk 3 above)
Comment (butonic) [reply pkg/service/v0/service.go]: The client will send the username, so we should store it by username for now. The client should be using the account id, but we need to clarify that in the next iteration of the url signing. I will create a ticket with followup tasks.

**Pull request merged**